modules/wallet/wallet.go

```diff
@@ -137,6 +139,29 @@ func New(cs modules.ConsensusSet, tpool modules.TransactionPool, persistDir stri
 	return w, nil
 }
 
+// Close terminates all ongoing processes involving the wallet, enabling
+// garbage collection.
+func (w *Wallet) Close() error {
+	var errs []error
+	// Lock the wallet outside of mu.Lock because Lock uses its own mu.Lock.
+	if w.Unlocked() {
+		if err := w.Lock(); err != nil {
+			errs = append(errs, err)
+		}
+	}
```

Comment:
This does not gaurantee that the wallet is locked at the time `Close()` returns as you could unlock the wallet in another goroutine after this block has executed, but before the `Lock` is obtained. This is probably extremely unlikely given the wallet's unlock time.

You could fix it with a `sync.ThreadGroup`, but I'm not sure that this is a semantic use of a `ThreadGroup`.